quickwit/quickwit-indexing/src/source/queue_sources/mod.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2024 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+mod local_state;
+#[cfg(test)]
+mod memory_queue;
+mod message;
+mod processor;
+mod shared_state;
+#[cfg(feature = "sqs")]
+mod sqs_queue;
+#[cfg(feature = "sqs")]
+pub mod sqs_source;
+mod visibility;
+
+use std::fmt;
+use std::time::{Duration, Instant};
+
+use async_trait::async_trait;
+use message::RawMessage;
+
+/// The queue abstraction is based on the AWS SQS and Google Pubsub APIs. The
+/// only requirement of the underlying implementation is that messages exposed
+/// to a given consumer are hidden to other consumers for a configurable period
+/// of time. Retries are
+#[async_trait]
+pub trait Queue: fmt::Debug + Send + Sync + 'static {
+    /// Poll the queue to receive messages.
+    ///
+    /// The implementation is in charge of choosing the wait strategy when there
+    /// are no messages in the queue. It will typically use long polling to do
+    /// this efficiently. On the other hand, when there is a message in the
```

Comment:
I didn't understand what was the meaning of the second setnence. If we don't long polling, we don't know when there is a message in the queue right away do we?

Reply:
I understood what you meant after reading other bits of code.